### PR TITLE
findex: init at 0.7.1

### PR DESCRIPTION
**Pull request by aquifolly**
###### Description of changes

https://github.com/mdgaziur/findex
#210519

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Hello, thanks for contributing ! In order to merge, we need to format the commit correctly first. Could you rework your branch into 2 commits : 
- maintainers: add pinkcreeper100
- findex: init at v0.7.1

Thanks !

**Comment by pinkcreeper100** [comment]
not exactly sure how to do that

**Comment by JulienMalka** [comment]
> not exactly sure how to do that
You probably want to do something like git reset HEAD~4, then commit again in the right order and by splitting the commits like explained, then push again using git push --force

**Comment by pinkcreeper100** [comment]
done

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Can you fix the conflicts ? (You want to rebase on master)

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Is this supposed to work on darwin ? If not, could you add a ``meta.platforms`` field ?

**Comment by pinkcreeper100** [comment]
do I squash the commit into the main one after I do that?

**Comment by JulienMalka** [comment]
> do I squash the commit into the main one after I do that?

Yes please :)

**Review by JulienMalka — APPROVED** [review]
Looks good to me :)

File: pkgs/applications/misc/findex/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, fetchFromGitHub, rustPlatform, pkg-config, keybinder3, gtk3 }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "findex";
+  version = "v0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "mdgaziur";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/findex/default.nix:10]:
```suggestion
  version = "0.7.1";

  src = fetchFromGitHub {
    owner = "mdgaziur";
    repo = pname;
    rev = "v${version}";
```

File: pkgs/applications/misc/findex/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  buildInputs = [
+    gtk3
+    keybinder3
+  ];
+
+  meta = with lib; {
+    description = "A highly customizable application finder written in Rust and uses Gtk3";
+    homepage = "https://github.com/mdgaziur/findex";
+    license = licenses.gpl3Only;
+    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux"];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/findex/default.nix:29]:
```suggestion
    platforms = platforms.linux;
```

Or simple for our usecases

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -38711,4 +38711,7 @@ with pkgs;
   resgate = callPackage ../servers/resgate { };
 
   oversteer = callPackage ../applications/misc/oversteer { };
+
+  findex = callPackage ../applications/misc/findex { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:38715]: This must be sorted.

File: pkgs/applications/misc/findex/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    pkg-config
+  ];
+
+  buildInputs = [
+    gtk3
+    keybinder3
+  ];
+
+  meta = with lib; {
+    description = "A highly customizable application finder written in Rust and uses Gtk3";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/findex/default.nix:26]:
```suggestion
    description = "Highly customizable application finder written in Rust and uses Gtk3";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (pinkcreeper100) [reply pkgs/top-level/all-packages.nix]: may I ask how?

**Comment by JulienMalka** [comment]
Result of `nixpkgs-review pr 211336` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>findex</li>
  </ul>
</details>

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Looks like your ``cargoHash`` is somehow wrong now.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (JulienMalka) [reply pkgs/top-level/all-packages.nix]: Just put this line in alphabetical order instead of at the end.

**Comment by pinkcreeper100** [comment]
> Looks like your `cargoHash` is somehow wrong now.

I think upstream made some changes

**Comment by JulienMalka** [comment]
Result of `nixpkgs-review pr 211336` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>findex</li>
  </ul>
</details>

**Review by JulienMalka — APPROVED** [review]

**Comment by pinkcreeper100** [comment]
where's the change that's requested?

**Comment by JulienMalka** [comment]
> where's the change that's requested?

I think it is fine, I don't have committer rights, so I approved the PR and we need a committer to merge it if they see fit.

**Comment by SuperSandro2000** [comment]
Please rebase

**Comment by pinkcreeper100** [comment]
@SuperSandro2000 done

**Comment by pinkcreeper100** [comment]
rebased to latest

**Pull request merged**